src/core/tree.ts

```diff
@@ -196,12 +196,14 @@ export class TreeNode<T>
     // TODO: how about children being Sets?
     children: TreeNode<T>[];
     data: T;
+    lastSelectedChild: NodeID | null;
```

Comment:
Aside for my hatred for `null`s -> this most definitely does not belong in the `core`, since it's a UI matter => but the same logic injected into UI Node would de most definitely sensible.